### PR TITLE
Allow project owner role to change site config #8005

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessRequiredException.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/processor/RoleContentProcessor.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5319660&bid=17409563)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessRequiredException.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/processor/RoleContentProcessor.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5319660&bid=17409563)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java  7
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessRequiredException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5319660&bid=17409563)

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java [hunk 1]
```diff
@@ -51,38 +49,33 @@ public ProcessUpdateResult processUpdate( final ProcessUpdateParams params )
...
+                }
+                else if ( !this.projectPermissionsContextManager.hasAnyProjectPermission( projectName, authenticationInfo,
+                                                                                          Set.of( ProjectRole.OWNER ) ) )
+                {
+                    throw new ProjectAccessRequiredException( authenticationInfo.getUser().getKey(), ProjectRole.OWNER );
+                }
+            }
         }
 
         return null;
```
Comment (anatol-sialitski) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java:73]: @vbradnitski why this method always returns null?

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManager.java [hunk 2]
```diff
@@ -22,5 +25,7 @@
 
     boolean hasAnyProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo );
 
+    boolean hasAnyProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo,
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManager.java:28]: hasAnyProjectRole and change arguments order

Move way all other has methods

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManagerImpl.java [hunk 3]
```diff
@@ -124,27 +124,18 @@ public boolean hasAnyProjectPermission( final ProjectName projectName, final Aut
...
     {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.AUTHOR ) );
+        return hasPermissions( projectName, authenticationInfo, projectRoles );
     }
 
-    private boolean hasContributorProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
+    private boolean hasOwnerProjectPermission( final ProjectName projectName, final AuthenticationInfo authenticationInfo )
     {
-        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.CONTRIBUTOR ) );
+        return hasPermissions( projectName, authenticationInfo, Set.of( ProjectRole.OWNER ) );
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManagerImpl.java:136]: We decided to use hasRole method of AuthInfo but it means that internally service must know how to construct project role names correctly - which should be done in another class (to share logic)

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManager.java [hunk 4]
```diff
@@ -1,7 +1,10 @@
 package com.enonic.xp.core.impl.project;
 
+import java.util.Set;
+
 import com.enonic.xp.context.Context;
 import com.enonic.xp.project.ProjectName;
+import com.enonic.xp.project.ProjectRole;
 import com.enonic.xp.security.auth.AuthenticationInfo;
 
 public interface ProjectPermissionsContextManager
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectPermissionsContextManager.java:10]: Maybe better to call it ProjectSecurityService and move away "context" logic

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessHelper.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java  7
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessRequiredException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5319660&bid=17409563)

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java (hunk 1 above)
Comment (vbradnitski) [reply modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java:73]: It's a processor, which called during create/update methods in a queue of similar classes. So it should check authInfo and throw an exception if it doesn't contain project role. Result of this particular processor is never used.

**Pull request merged**